### PR TITLE
Animations & CustomFOV: much more customization

### PR DESCRIPTION
**Pull request by Kefpull**
Lots of changes to Animations.java

added StartingSwingProgress which affects blockModes none, smooth, exhibition, spin, sigma, wood, allah
moved swingSpeed here

>new categories & settings
CustomTranslation (pre): translation without affecting the block animations
CustomScaling: custom x,y,z scaling for the held item
CustomRotation: custom x,y,z rotation for the held item

>new module
CustomFOV
ability to set between 1-179 degrees
staticFOV: emulates optifine dynamic fov off

+ fix for intelliJ recognizing the blocked out function lines as errors

## Summary by CodeRabbit

- **New Features**
	- Introduced a `modifyAnimations` setting for enhanced animation customization.
	- Added new settings for translation, scaling, and rotation adjustments.
	- Users can now toggle custom scaling and rotation features.
	- Implemented a new `CustomFOV` module for customizable Field of View settings.
	- Added a new event for handling FOV updates, allowing real-time adjustments.

- **Improvements**
	- Enhanced control flow for animation rendering, allowing for dynamic adjustments based on user preferences.
	- Improved functionality for dynamic FOV adjustments based on user settings.

These updates provide users with greater flexibility and control over animation and FOV settings, significantly improving the overall user experience.

### CODE REVIEW

**Comment by Kefpull** [comment]
@coderabbitai review

File: src/main/java/keystrokesmod/module/impl/render/Animations.java [hunk 1]
```diff
@@ -280,19 +366,63 @@
...
     private void translate(double x, double y, double z) {
         GlStateManager.translate(
-                x + this.x.getInput(),
-                y + this.y.getInput(),
-                z + this.z.getInput()
+                x + this.translatex.getInput(),
+                y + this.translatey.getInput(),
+                z + this.translatez.getInput()
+        );
+    }
```
[the remark this answers is not in the document]
Comment (Kefpull) [reply src/main/java/keystrokesmod/module/impl/render/Animations.java:381]: translate is not only used to perform the base translation of an object but is also used in block animations to adjust the item to the correct position, its not double addition most of the time

**Comment by Kefpull** [comment]
maybe it would be good to rename translate() to translateWithOffset() and just use the base function from gl11 to do the pre translation

**Comment by Kefpull** [comment]
@coderabbitai full review

**Review by xia-mc — COMMENTED** [review]
I will delay this pull request to 2.14, because i can't test it now.

**Comment by xia-mc** [comment]
> @xia-mc updated animations with swingspeedwhileblocking for merge

Thanks

**Pull request merged**